influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.influxdb;
+
+import com.hazelcast.jet.function.QuadFunction;
+import com.hazelcast.jet.function.SupplierEx;
+import com.hazelcast.jet.pipeline.BatchSource;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.SourceBuilder.SourceBuffer;
+import org.influxdb.InfluxDB;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.influxdb.impl.InfluxDBResultMapper;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.Consumer;
+
+import static com.hazelcast.util.Preconditions.checkTrue;
+import static java.util.Objects.nonNull;
+import static org.influxdb.InfluxDBFactory.connect;
+
+/**
+ * Contains factory methods for creating InfluxDB sources
+ */
+public final class InfluxDbSources {
+
+    /**
+     * Default number of {@link QueryResult}s to process in one chunk
+     */
+    public static final int DEFAULT_CHUNK_SIZE = 100;
+
+    private InfluxDbSources() {
+
+    }
+
+    /**
+     * Creates a source that executes the query on given database and
+     * emits items mapped with user defined mapper function.
+     * Authenticates with the server using given credentials.
+     *
+     * @param query             query to execute on InfluxDb database
+     * @param database          name of the database
+     * @param url               url of the InfluxDb server
+     * @param username          username of the InfluxDb server
+     * @param password          password of the InfluxDb server
+     * @param chunkSize         the number of {@link QueryResult}s to process in one chunk.
+     * @param measurementMapper mapper function which takes measurement name, tags set, column names and row
+     *                          as argument and produces the user object {@link T} which will be emitted from
+     *                          this source
+     * @param <T>               type of the user object
+     * @return {@link T} mapped user objects
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query, @Nonnull String database, @Nonnull String url,
```

Comment:
* I don't think `chunkSize` needs to be configurable - it should be set to a reasonable default
* Instead of QuadFunction we should use a custom interface that's more user friendly